tests/rendered_output.rs

```diff
@@ -339,3 +339,101 @@ fn book_with_a_reserved_filename_does_not_build() {
+        let temp = DummyBook::new().build().unwrap();
+        let md = MDBook::load(temp.path()).unwrap();
+        md.build().unwrap();
+
+        let index = temp.path().join("book/searchindex.json");
+        let index = file_to_string(index).unwrap();
+        let index: serde_json::Value = serde_json::from_str(&index).unwrap();
+
+        let bodyidx = &index["index"]["index"]["body"]["root"];
+        let textidx = &bodyidx["t"]["e"]["x"]["t"];
+        assert_eq!(textidx["df"], 2);
+        assert_eq!(textidx["docs"]["first/index.html#first-chapter"]["tf"], 1.0);
+        assert_eq!(textidx["docs"]["intro.html#introduction"]["tf"], 1.0);
+
+        let docs = &index["index"]["documentStore"]["docs"];
+        assert_eq!(docs["first/index.html#first-chapter"]["body"], "more text.");
+        assert_eq!(docs["first/index.html#some-section"]["body"], "");
+        assert_eq!(
+            docs["first/includes.html#summary"]["body"],
+            "Introduction First Chapter Nested Chapter Includes Second Chapter Conclusion"
+        );
+        assert_eq!(
+            docs["first/includes.html#summary"]["breadcrumbs"],
+            "First Chapter » Summary"
+        );
+        assert_eq!(
+            docs["conclusion.html#conclusion"]["body"],
+            "1 + 2 < 5 > 0 &gt;"
+        );
+    }
+
+    // Setting this to `true` may cause issues with `cargo watch`,
+    // since it may not finish writing the fixture before the tests
+    // are run again.
+    const GENERATE_FIXTURE: bool = false;
+
+    fn get_fixture() -> serde_json::Value {
+        if GENERATE_FIXTURE {
+            let temp = DummyBook::new().build().unwrap();
+            let md = MDBook::load(temp.path()).unwrap();
+            md.build().unwrap();
+
+            let src = temp.path().join("book/searchindex.json");
+            let src = File::open(src).unwrap();
+            let src: serde_json::Value = serde_json::from_reader(src).unwrap();
+
+            let dest = Path::new(env!("CARGO_MANIFEST_DIR")).join("tests/searchindex_fixture.json");
+            let mut dest = File::create(&dest).unwrap();
+            serde_json::to_writer_pretty(&mut dest, &src).unwrap();
+
+            src
+        } else {
+            let json = include_str!("searchindex_fixture.json");
+            serde_json::from_str(json).expect("Unable to deserialize the fixture")
+        }
+    }
+
+    // So you've broken the test. If you changed dummy_book, it's probably
+    // safe to regenerate the fixture. If you haven't then make sure that the
+    // search index still works. Run `cargo run -- serve tests/dummy_book`
+    // and try some searches. Are you getting results? Do the teasers look OK?
+    // Are there new errors in the JS console?
+    //
+    // If you're pretty sure you haven't broken anything, change `GENERATE_FIXTURE`
+    // above to `true`, and run `cargo test` to generate a new fixture. Then
+    // change it back to `false`.
```

Comment:
I got a nice laugh out of this block of comments, yet it's still surprisingly useful for people who see it because of a failing test. Thank you!